packages/harmony/src/components/text/Text/Text.tsx

```diff
@@ -1,48 +1,108 @@
 
+import { CSSObject, useTheme } from '@emotion/react'
 import { Slot } from '@radix-ui/react-slot'
-import cn from 'classnames'
-import { camelCase } from 'lodash'
 
-import { variantTagMap } from './constants'
+import { typography } from 'foundations'
+
 import type { TextProps } from './types'
-import styles from './typography.module.css'
+
+const variantTagMap = {
+  display: {
+    xl: 'h1',
+    l: 'h1',
+    m: 'h1',
+    s: 'h1'
+  },
+  heading: {
+    xl: 'h1',
+    l: 'h2',
+    m: 'h3',
+    s: 'h4'
+  }
+}
+
+const variantStylesMap = {
+  display: {
+    fontSize: { s: '6xl', m: '7xl', l: '8xl', xl: '9xl' },
+    lineHeight: { s: '3xl', m: '4xl', l: '5xl', xl: '6xl' },
+    fontWeight: { default: 'bold', strong: 'heavy' }
+  },
+  heading: {
+    fontSize: { s: 'xl', m: '2xl', l: '3xl', xl: '3xl' },
+    lineHeight: { s: 'l', m: 'xl', l: 'xl', xl: 'xl' },
+    fontWeight: { default: 'bold', strong: 'heavy' }
+  },
+  title: {
+    fontSize: { xs: 'xs', s: 's', m: 'm', l: 'l' },
+    lineHeight: { xs: 's', s: 's', m: 'm', l: 'l' },
+    fontWeight: { weak: 'demiBold', default: 'bold', strong: 'heavy' }
+  },
+  label: {
+    fontSize: { xs: '2xs', s: 'xs', m: 's', l: 'm' },
+    lineHeight: { xs: 's', s: 's', m: 'm', l: 'm' },
+    fontWeight: { default: 'bold', strong: 'heavy' }
+  },
+  body: {
+    fontSize: { xs: 'xs', s: 's', m: 'm', l: 'l' },
+    lineHeight: { xs: 's', s: 'm', m: 'm', l: 'l' },
+    fontWeight: { default: 'medium', strong: 'demiBold' }
+  }
+}
 
 export const Text = forwardRef(
   <TextComponentType extends ElementType = 'p'>(
     props: TextProps<TextComponentType>,
     ref: ForwardedRef<TextComponentType>
   ) => {
     const {
-      className,
       children,
       variant,
       strength = 'default',
       size = 'm',
       color,
       tag,
-      innerRef,
       asChild,
       ...other
     } = props
 
-    const Tag: ElementType = asChild
-      ? Slot
-      : tag ?? variantTagMap[variant ?? 'body'][size] ?? 'p'
+    const theme = useTheme()
+
+    const variantConfig = variant && variantStylesMap[variant]
+
+    const styles: CSSObject = {
+      fontFamily: `'Avenir Next LT Pro', 'Helvetica Neue', Helvetica,
+    Arial, sans-serif`,
+      position: 'relative',
+      boxSizing: 'border-box',
+      ...(color &&
+        color === 'heading' && {
+          color: theme.color.secondary.secondary,
+          WebkitTextFillColor: 'transparent',
+          backgroundClip: 'text',
+          backgroundImage: theme.color.text.heading
+        }),
+      ...(color && color !== 'heading' && { color: theme.color.text[color] }),
+      ...(variantConfig && {
+        // @ts-expect-error
+        fontSize: typography.size[variantConfig.fontSize[size]],
+        // @ts-expect-error
+        lineHeight: typography.lineHeight[variantConfig.lineHeight[size]],
+        // @ts-expect-error
+        fontWeight: typography.weight[variantConfig.fontWeight[strength]]
```

Comment:
What are the errors happening here?

Reply:
Yeah kinda sucks. So the size prop does not always match with the allowed sizes for some variants. The fix could maybe be having only some sizes based on the variant, but would be a lot to type

Reply:
Ahh right. Yea we can take a look at this again later. 👍